### DISCUSSION TITLE
Poor speed using wifi 6 esp32c6

### DISCUSSION THREAD

**Discussion by Dayzen** [dchristl/esp32_nat_router_extended#137] [General]
Hello, I'm using your solution on esp32c6 which is wifi 6 capable. Direct connection to wifi source gives me something like 120-150 Mbps on speedtest while connecting thru esp32c6 AP (stationned on same network of course, and doing speedtest 20cm away from the esp) I only get 16Mbps max. 

ChatGpt give me them clues that could seem relevant:
- Channel Bandwidth: Ensure that the ESP32-C6 is using a 40 MHz or 80 MHz channel bandwidth. A narrower bandwidth (e.g., 20 MHz) would significantly limit speed.
- Operating Mode: The ESP32-C6 might not be configured to operate in full Wi-Fi 6 mode, or it could be limited by firmware or software settings.
- 1024-QAM (Quadrature Amplitude Modulation) can significantly boost data rate. But if signal strength or quality is low, the module might fall back to a lower QAM level (e.g., 256-QAM, 64-QAM), resulting in reduced speeds.
- Optimizing dBm: The default transmit power for the ESP32-C6 might be set lower to save power or reduce interference. You can adjust this in the firmware or via software to ensure it is set to the maximum allowable level for your situation, typically around 20 dBm for 2.4 GHz Wi-Fi. If the power is too low, it can severely limit speed, even at close range.

Thanks in advance for your concerns on this subject

**Comment by dchristl** [comment]
Hello,

the firmware for the ESP32 is the same for all variants and does not contain any optimisations for specific variations. In this way, I keep the administration and further development effort to a minimum. Furthermore, I do not own a C6 and have only compiled this at the request of a user (see #105 ).

Kind Regards,
Danny
